### ISSUE TITLE
Performance analysis: parent task dependency

### ISSUE THREAD

**Issue by nicelhc13** [ut-parla/parla-experimental#78]
Compared to thread wait(), it is trivial for now but depending on workloads, it could be bottleneck.
![analysis2_3gpus](https://user-images.githubusercontent.com/12228132/231391619-0f2f376b-55f1-484f-9698-918cb2bfb984.PNG)
The above is 16ms/1000 independent task results.
As you can see, the marked part took 120ms. If we consider a task is designed to run 16ms, this is nontrivial. 
This was because all child tasks (independent tasks) call `process_dependencies()` in parallel and this causes thread conflicts actively.

One possible idea could be to make dependencies from the parent task, instead of child tasks.
This is low priority disucssion.
